Bio/SeqFeature.py

```diff
@@ -1188,6 +1198,24 @@ def __iter__(self):
             for pos in loc:
                 yield pos
 
+    def __eq__(self, other):
+        """Check if all parts of CompoundLocation are equal to all parts of other CompoundLocation"""
+        if not isinstance(other, CompoundLocation):
+            return False
+        if not len(self.parts) == len(other.parts):
```

Comment:
Was this deliberate over using ``if len(self.parts) != len(other.parts):`` (and likewise for the rest of the method)?